app/src/main/java/com/github/h3lp3rs/h3lp/locationmanager/LocationHelper.kt

```diff
@@ -0,0 +1,53 @@
+import android.content.Context
+import android.content.Intent
+import android.location.Location
+import android.location.LocationManager
+import com.github.h3lp3rs.h3lp.MainPageActivity
+import com.github.h3lp3rs.h3lp.locationmanager.GeneralLocationManager
+
+
+/**
+ * SuperActivity that is common to all activities using localization
+ */
+class LocationHelper {
+    private var userLocation: Location? = null
+
+    fun getUserLatitude(): Double?{
+        return userLocation?.latitude
+    }
+
+    fun getUserLongitude(): Double?{
+        return userLocation?.longitude
+    }
+
+    /**
+     * Function that updates the user's current coordinates
+     * @param context: the context of the activity using the coordinates
+     */
+    fun updateCoordinates(context: Context) {
+        val futureLocation = GeneralLocationManager.get().getCurrentLocation(context)
+        futureLocation.thenAccept {
+            userLocation = Location(LocationManager.GPS_PROVIDER)
```

Comment:
I guess it'd be more accurate to say that it's a Location(LocationManager.FUSED_PROVIDER) but then this would apparently add a requiresApi section on every method using the LocationHelper :/ GPS_PROVIDER should be fine